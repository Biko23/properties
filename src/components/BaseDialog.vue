<template>
    <v-dialog transition="dialog-top-transition" persistent :value="dialogState" max-width="600">
        <template>
            <v-card>
                <v-toolbar color="deep-purple accent-4" dark>{{title}}</v-toolbar>
                <v-card-text class="pt-5">
                    <p style="font-size: 16px">{{ message }}</p>
                </v-card-text>
                <v-card-actions class="justify-end">
                    <slot name="button" />
                </v-card-actions>
            </v-card>
        </template>
    </v-dialog>
</template>
<script>
    export default {
        name: "BaseDialog",
        props: {
            message: {
                type: String,
                default: "No message"
            },
            title: {
                type: String,
                default: "No title"
            },
            dialogState: {
                type: Boolean,
                default: false
            }
        }
    }
</script>