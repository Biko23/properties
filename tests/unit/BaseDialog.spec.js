import { createLocalVue, mount } from '@vue/test-utils'
import Vuetify from 'vuetify'
import BaseDialog from '@/components/BaseDialog.vue'

describe('BaseDialog.vue', () => {
  const localVue = createLocalVue()
  let vuetify

  beforeEach(() => {
    vuetify = new Vuetify()
  })

  const mountFunction = options => {
    return mount(BaseDialog, {
      localVue,
      vuetify,
      ...options,
    })
  }

  it('should have a match snapshot', () => {
    const wrapper = mountFunction()

    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot()

    // We could also verify this differently
    // by checking the text content
    // const title = wrapper.find('.v-card__title > span')
    // expect(title.text()).toBe('Foobar')
  })

  // it('should have default props details', () => {
  //       const wrapper = mountFunction()
  //       expect(wrapper.find('[data-testid="dialog-state-element"]').props().value).toBe(false)
  //       expect(wrapper.find('[data-testid="title-element"]').text()).toBe('No message')
  //       expect(wrapper.find('[data-testid="message-element"]').text()).toBe(null)
  //   })

  // it('should have modified props details', () => {
  //     const wrapper = mountFunction({
  //         propsData: { 
  //           dialogState: true,
  //           title: 'Success',
  //           message: 'Succeeded'
  //         }
  //       })

  //       expect(wrapper.find('[data-testid="dialog-state-element"]').props().value).toBe(true)
  //       expect(wrapper.find('[data-testid="title-element"]').text()).toBe('Success')
  //       expect(wrapper.find('[data-testid="message-element"]').text()).toEqual('Succeeded')
  // })
})

