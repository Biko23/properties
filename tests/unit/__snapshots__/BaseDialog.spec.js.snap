// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseDialog.vue should have a match snapshot 1`] = `
<div role="dialog" class="v-dialog__container" data-testid="dialog-state-element">
  <!---->
</div>
`;
